pkg/operator/controller/gatewayclass/controller.go

```diff
@@ -90,15 +123,29 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 
 	var gatewayclass gatewayapiv1.GatewayClass
 	if err := r.cache.Get(ctx, request.NamespacedName, &gatewayclass); err != nil {
+		log.Error(err, "failed to get gatewayclass", "request", request)
 		return reconcile.Result{}, err
 	}
 
 	var errs []error
 	if _, _, err := r.ensureServiceMeshOperatorSubscription(ctx); err != nil {
+		log.Error(err, "failed to ensure ServiceMeshOperatorSubscription", "request", request)
 		errs = append(errs, err)
 	}
 	if _, _, err := r.ensureServiceMeshControlPlane(ctx, &gatewayclass); err != nil {
+		log.Error(err, "failed to ensure ServiceMeshControlPlane", "request", request)
 		errs = append(errs, err)
+	} else {
+		r.startSMCPWatch.Do(func() {
```

Comment:
Why would we want to start a watch every time we reconcile?

Reply:
Hey @candita, We are using sync.Once here, start the watch for SMCP, which should run only once. So, this should not run on every reconciliation.

Since we need to create the SMCP watch after the OSSM operator is installed, we cannot start the watch as usual.